File: library/src/main/java/com/github/lany192/box/mvp/BaseView.java
package com.github.lany192.box.mvp;

import androidx.lifecycle.LifecycleOwner;

public interface BaseView extends LifecycleOwner {

    void showEmpty();

    void showEmpty(String msg);

    void showContent();

    void showNoWifi();

    void showError();

    void showError(String msg);

    void showLoading();

    void showLoadingDialog();

    void showLoadingDialog(CharSequence message);

    void cancelLoadingDialog();
}

File: library/src/main/java/com/github/lany192/box/mvp/BaseView.kt
package com.github.lany192.box.mvp

import androidx.lifecycle.LifecycleOwner

interface BaseView : LifecycleOwner {
    fun showEmpty()
    fun showEmpty(msg: String?)
    fun showContent()
    fun showNoWifi()
    fun showError()
    fun showError(msg: String?)
    fun showLoading()
    fun showLoadingDialog()
    fun showLoadingDialog(message: CharSequence?)
    fun cancelLoadingDialog()
}